alignak_backend/livesynthesis.py

```diff
@@ -62,120 +69,157 @@ def recalculate():
             # Update hosts live synthesis
             hosts = current_app.data.driver.db['host']
             hosts_count = hosts.find({'_is_template': False, '_realm': realm['_id']}).count()
-            if live_current['hosts_total'] != hosts_count:
-                data = {"hosts_total": hosts_count}
-
-                data['hosts_up_hard'] = hosts.find({
-                    '_is_template': False,
-                    "ls_state": "UP", "ls_state_type": "HARD",
-                    "ls_acknowledged": False, "_realm": realm["_id"]
-                }).count()
-                data['hosts_down_hard'] = hosts.find({
-                    '_is_template': False,
-                    "ls_state": "DOWN", "ls_state_type": "HARD",
-                    "ls_acknowledged": False, "_realm": realm["_id"]
-                }).count()
-                data['hosts_unreachable_hard'] = hosts.find({
-                    '_is_template': False,
-                    "ls_state": "UNREACHABLE", "ls_state_type": "HARD",
-                    "ls_acknowledged": False, "_realm": realm["_id"]
-                }).count()
-
-                data['hosts_up_soft'] = hosts.find({
-                    '_is_template': False,
-                    "ls_state": "UP", "ls_state_type": "SOFT",
-                    "ls_acknowledged": False, "_realm": realm["_id"]
-                }).count()
-                data['hosts_down_soft'] = hosts.find({
-                    '_is_template': False,
-                    "ls_state": "DOWN", "ls_state_type": "SOFT",
-                    "ls_acknowledged": False, "_realm": realm["_id"]
-                }).count()
-                data['hosts_unreachable_soft'] = hosts.find({
-                    '_is_template': False,
-                    "ls_state": "UNREACHABLE", "ls_state_type": "SOFT",
-                    "ls_acknowledged": False, "_realm": realm["_id"]
-                }).count()
-
-                data['hosts_acknowledged'] = hosts.find({
-                    '_is_template': False,
-                    'ls_acknowledged': True, "_realm": realm["_id"]
-                }).count()
-                data['hosts_in_downtime'] = hosts.find({
-                    '_is_template': False, 'ls_downtimed': True, "_realm": realm["_id"]
-                }).count()
-
-                lookup = {"_id": live_current['_id']}
-                patch_internal('livesynthesis', data, False, False, **lookup)
+            data = {'hosts_total': hosts_count}
+
+            data['hosts_not_monitored'] = hosts.find({
+                '_is_template': False,
+                'active_checks_enabled': False, 'passive_checks_enabled': False,
+                '_realm': realm['_id']
+            }).count()
+            data['hosts_up_hard'] = hosts.find({
+                '_is_template': False,
+                '$or': [{'active_checks_enabled': True}, {'passive_checks_enabled': True}],
+                'active_checks_enabled': False, 'passive_checks_enabled': False,
```

Comment:
You have the `$or` the line before so not need have this line `'active_checks_enabled': False, 'passive_checks_enabled': False,`

Reply:
+1 for you 😉